libraries/rodeos/include/b1/rodeos/callbacks/chaindb.hpp

```diff
@@ -0,0 +1,882 @@
+#pragma once
+
+#include <b1/rodeos/callbacks/kv.hpp>
+#include <b1/rodeos/constants.hpp>
+#include <eosio/ship_protocol.hpp>
+#include <eosio/to_key.hpp>
+
+namespace b1::rodeos
+{
+   using contract_index_double = eosio::ship_protocol::contract_index_double;
```

Comment:
I suspect it's not with the development cost. Most of the ecosystem outside of nodeos can't handle it. eosiolib's abi serialization does json conversion using double as an intermediary. eosjs represents it as hex bytes. Because of that, I doubt many contracts use it.